docs/_chapters/230-manifest-annotations.md

```diff
@@ -1,136 +1,286 @@
 
-See the Javadoc for additional details.
+### Custom Bundle Annotations
 
-### Example
+Certain *bundle annotations* have a second important use. We know that if applied to a type or package *bundle annotations* result in a clause in the manifest. However, many can be used as meta-annotations to a second annotation. The second annotation is considered a ***custom bundle annotation***. The *custom bundle annotation* results in a manifest clause only when applied to a type or package.
 
-	@ASL_2_0
-	public class Peggy { ... }
+This makes it possible to create an annotation for a subsystem. For example, an annotation `@ASL_2_0` that sets the `Bundle-License` header to the Apache Software License version 2.0.
 
-	Bundle-License: \
-		http://www.opensource.org/licenses/apache2.0.php; \
-		description='Apache Software License 2.0';\
-		link='http://www.apache.org/licenses/LICENSE-2.0.html'
+```java
+@BundleLicense(
+    name = "http://www.opensource.org/licenses/apache2.0.php", 
+    link = "http://www.apache.org/licenses/LICENSE-2.0.html", 
+    description = "Apache Software License 2.0")
+@interface ASL_2_0 {}
 
-## More Manifest Annotations
+// takes effect when applied to a type
+
+@ASL_2_0
+class Foo { ... }
+```
+
+### Adding Attributes and Directives
+
+When creating *custom bundle annotations* a common requirement is to make them parameterizable such that the values of the *custom bundle annotation* feed into the header clauses resulting from the *bundle annotation* applied to it (*remember; a **custom bundle annotation** is meta-annotated with a **bundle annotation***.)
+
+[OSGi](https://osgi.org/specification/osgi.core/7.0.0/) specifies two annotations, [`@Attribute`](https://osgi.org/specification/osgi.core/7.0.0/framework.api.html#org.osgi.annotation.bundle.Attribute) and [`@Directive`](https://osgi.org/specification/osgi.core/7.0.0/framework.api.html#org.osgi.annotation.bundle.Directive), for this purpose. Any methods of the **custom bundle annotation** annotated with `@Attribute` or  `@Directive` will result in those becoming additional attributes or directives respectively of the resulting header clause _when a value is supplied_.
+
+##### `@Attribute`
+
+`@Attribute` allows you to add new or update existing attributes from the _bundle annotation_.
+
+```java
+@Capability(namespace = "foo.namespace")
+@interface Extended {
+	@Attribute("foo.attribute") // this attribute enhances the @Capability
+	String value();
+}
+
+// usage
+
+@Extended("bar")
+class Foo {}
+```
+
+which results in the manifest header:
+
+`Provide-Capability: foo.namespace;foo.attribute=bar`
+
+##### `@Directive`
+
+`@Directive` behaves similarly; with some caveats. You can add new or update existing directives for _non-specified namespaces_.
+
```

Comment:
What do you mean by "non-specified namespaces"? Is this because the Verifier will complain? People can elide the error with `-fixupmessages`.

Reply:
FWIW `-fixupmessages` is not a viable solution for a reusable annotation. You expect people to first use the annotation, and then have to cleanup errors with `-fixupmessages`?